script.artistslideshow/default.py

```diff
@@ -0,0 +1,1153 @@
+
+    def __init__( self, workqueue, queuelock, window, delay, fadetoblack ):
+        super( Slideshow , self).__init__()
+        self.WORKQUEUE = workqueue
+        self.QUEUELOCK= queuelock
+        self.WINDOW = window
+        self.DELAY = delay
+        self.IMAGES = []
+        self.IMAGEADDED = False
+        self.IMAGESCLEARED = False
+        self.SHOW = True
+        self.FADETOBLACK = fadetoblack
+        lw.log( ['Slideshow thread started'] )
+
+
+    def AddImage( self, path ):
+        if path:
+            self.IMAGES.append( path )
+            lw.log( ['Added to image display group: ' + path] )
+            self.IMAGEADDED = True
+            return True
+        else:
+            lw.log( ['Image path was empty, nothing added'] )
+            return False
+
+
+    def ClearImages( self ):
+        self.IMAGES = []
+        if self.FADETOBLACK:
+            self._set_property( 'ArtistSlideshow.Image', os.path.join( addonpath, 'resources', 'images', 'black-hd.png' ) )
+        else:
+            self._set_property( 'ArtistSlideshow.Image' )
+        lw.log( ['images cleared'] )
+        self.IMAGESCLEARED = True
+
+
+    def run( self ):
+        cmd = ''
+        last_image = ''
+        if self.FADETOBLACK:
+            self._set_property( 'ArtistSlideshow.Image', os.path.join( addonpath, 'resources', 'images', 'black-hd.png' ) )
+        while self.SHOW:
+            outofimages = True
+            with self.QUEUELOCK:
+                if not self.WORKQUEUE.empty():
+                    cmd = self.WORKQUEUE.get()
+            if cmd == 'quit':
+                break
+            if self.IMAGEADDED or self.IMAGESCLEARED or outofimages:
+                random.shuffle( self.IMAGES )
+#                lw.log( ['the images available are', self.IMAGES] )
+                self.IMAGEADDED = False
+                self.IMAGESCLEARED = False
+                outofimages = False
+            for image in self.IMAGES:
+                if self.IMAGEADDED or self.IMAGESCLEARED:
+                    lw.log( ['image list changed, resetting loop'] )
+                    break
+                if not image == last_image:
+                    self._set_property( 'ArtistSlideshow.Image', image )
+                    lw.log( ['ArtistSlideshow.Image set to ' + image] )
+                    self._wait( self.DELAY )
+                    last_image = image
+                if not self.SHOW:
+                    break
+        lw.log( ['Slideshow thread stopping'] )
+
+
+    def _set_property( self, property_name, value="" ):
+        try:
+          self.WINDOW.setProperty( property_name, value )
+          lw.log( ['%s set to %s' % (property_name, value)] )
+        except Exception as e:
+          lw.log( ["Exception: Couldn't set propery " + property_name + " value " + value , e])
+
+
+    def _wait( self, wait_time ):
+        cmd = ''
+        waited = 0
+        while( waited < wait_time ):
+            time.sleep(0.1)
```

Comment:
I'm sorry I didn't spot this early. Please use xbmc.sleep(x) for times lower than 5 secs and xbmc.Monitor().waitForAbort(x) otherwise. This extends to any usage of time.sleep in the addon